anvil/config.py

```diff
@@ -185,15 +185,36 @@ def parse_bootstrap_sample_size(self, n_boot: int):
         log.warning(f"Using user specified bootstrap sample size: {n_boot}")
         return n_boot
 
-    def produce_bootstrap_seed(
-        self, manual_bootstrap_seed: (int, type(None)) = None):
+    def produce_bootstrap_seed(self, manual_bootstrap_seed: (int, type(None)) = None):
         if manual_bootstrap_seed is None:
             return randint(0, maxsize)
         # numpy is actually this strict but let's keep it sensible.
-        if (manual_bootstrap_seed < 0) or (manual_bootstrap_seed > 2**32):
+        if (manual_bootstrap_seed < 0) or (manual_bootstrap_seed > 2 ** 32):
             raise ConfigError("Seed is outside of appropriate range: [0, 2 ** 32]")
         return manual_bootstrap_seed
 
+    def parse_cosh_fit_min_separation(self, n: int, training_geometry):
```

Comment:
can this not just take the `lattice_length` directly?

Reply:
yep, good point

Reply:
I can replace `training_geometry.length` with `lattice_length` in this parse function, but not the production rule directly below. I'm sure by now I should understand this behaviour but alas, I forget.

Also, since we're moving towards removing `training_context` (which contains the entire training runcard) in favour of things more akin to `training_geometry` which just extract the specific parameters we need, perhaps it does make sense to leave this as is.

Reply:
that's really weird, I don't understand that..

Reply:
I guess leave it for now.

Reply:
apologies, it doesn't work with either the parse or produce - the parse just worked because I was using the default value for `cosh_fit_min_separation` in `cosh_fit_window`..

Reply:
I think it's because that key is only present in the training context and so we could in theory put lattice length but we'd have to collect over the fit or write the key in the sampling runcard, idk why I didn't think about this earlier.

So definitely leave it for now and I will look into extracting parameters from the trained model in a more satisfactory way..

Reply:
aka the issue you linked.